torch/testing/_internal/common_methods_invocations.py

```diff
@@ -999,12 +1008,31 @@ def sample_inputs_linspace(op, device, dtype, requires_grad, **kwargs):
     for start, end, nstep in cases:
         if dtype == torch.uint8 and end < 0 or start < 0:
             continue
-        yield SampleInput(start, args=(end, nstep), kwargs={"dtype": dtype, "device": device})
+
+        tensor_options = {"dtype": dtype, "device": device}
+
+        yield SampleInput(start, args=(end, nstep), kwargs=tensor_options)
+        yield SampleInput(
+            torch.tensor(
+                start,
+                dtype=torch.float32 if isinstance(start, float) else torch.int64,
+                device=device
+            ),
+            args=(
+                torch.tensor(
+                    end,
+                    dtype=torch.float32 if isinstance(end, float) else torch.int64,
+                    device=device
+                ),
+                nstep
+            ),
+            kwargs=tensor_options
+        )
 
     yield SampleInput(1, args=(3, 1))
 
 
-def sample_inputs_logpace(op, device, dtype, requires_grad, **kwargs):
+def sample_inputs_logspace(op, device, dtype, requires_grad, **kwargs):
```

Comment:
Could we add new and separate OpInfos for linspace and logspace with Tensor inputs? This will make it easier to tell if we have regressed something in the existing testing

Reply:
This sounds a bit weird. Do you mean that we add a new `OpInfo` identical to the current one except the `sample_inputs_*space` function is different (for testing the Tensor inputs)?

Reply:
Yes

Reply:
If we add a new `OpInfo` with the same tested op the following line complains:

https://github.com/pytorch/pytorch/blob/35cca799ff42182a1b7f1ee4d0225ee879b7c924/torch/testing/_internal/common_device_type.py#L435

Reply:
You can have two OpInfos for the same function, you just need to add `variant_test_name` so the generated tests have different names.

Reply:
@peterbell10 thanks, done.